apps/wallet/src/features/Swap/InputSwap/ToToken.tsx

```diff
@@ -0,0 +1,101 @@
+import { useEffect } from 'react';
+import { StyleSheet } from 'react-native';
+import { Text, View } from '@walless/gui';
+import type { TokenDocument } from '@walless/store';
+import { showError } from 'modals/Error';
+import { useDebouncedCallback } from 'use-debounce';
+import { parseWithDecimals } from 'utils/format';
+import type { JupiterToken } from 'utils/hooks';
+import { useSnapshot } from 'utils/hooks';
+import { getAliasedMint, getSwapQuote } from 'utils/transaction';
+
+import type { SwapContext } from '../context';
+import { swapActions, swapContext } from '../context';
+
+import SelectButton from './SelectButton';
+
+const ToToken = () => {
+	const { fromToken, toToken, amount, swapQuote } = useSnapshot(swapContext)
+		.swap as SwapContext;
+
+	const handleSelectToToken = () => {
+		swapActions.openSelectToken('to');
+	};
+
+	const outAmount =
+		swapQuote && toToken
+			? parseWithDecimals(swapQuote.outAmount, toToken.decimals)
+			: 0;
+
+	const updateSwapQuote = useDebouncedCallback(
+		async (
+			fromToken?: TokenDocument,
+			toToken?: JupiterToken,
+			amount?: string,
+		) => {
+			if (!fromToken || !toToken || !amount) {
+				swapActions.update({ swapQuote: undefined });
+				return;
+			}
+
+			const amountValue = parseFloat(amount);
+			if (isNaN(amountValue) || amountValue === 0) {
+				swapActions.update({ swapQuote: undefined });
+				return;
+			}
+
+			const swapQuote = await getSwapQuote({
+				fromMint: getAliasedMint(fromToken),
+				toMint: toToken.address,
+				amount: amountValue * 10 ** fromToken.account.decimals,
+			});
+			if (!swapQuote) {
+				showError(
+					{ errorText: 'Can not swap these tokens, try another one' },
+					1500,
+				);
+			} else {
+				swapActions.update({ swapQuote });
+			}
+		},
+		1000,
+	);
+
+	useEffect(() => {
+		updateSwapQuote(fromToken, toToken, amount);
+	}, [fromToken, toToken, amount]);
+
+	return (
+		<View style={styles.container}>
+			<Text style={styles.title}>To</Text>
+			<View style={styles.tokenContainer}>
+				<SelectButton
+					symbol={toToken?.symbol}
+					logoURI={toToken?.logoURI}
+					onPress={handleSelectToToken}
+				/>
+				<Text style={styles.outAmount}>{outAmount}</Text>
+			</View>
+		</View>
+	);
+};
+
+export default ToToken;
+
+const styles = StyleSheet.create({
+	container: {
+		gap: 14,
+		paddingBottom: 30,
+	},
+	title: {
+		color: '#FFFFFF',
+	},
+	tokenContainer: {
+		flexDirection: 'row',
+		justifyContent: 'space-between',
+	},
+	outAmount: {
```

Comment:
may rename it to `outAmountText`

Reply:
it is not necessary to rename, `name` and `type` for this purpose, we just need to add suffix like `text` in case it is confusing or has the same name with a different var